### PR TITLE
Make File Dialog and Adventure Dialog buttons translatable

### PR DESCRIPTION
**Pull request by zenseii**
Both Evil and Good interfaces.

The cancel button will be handled later.

Relates to #6105 

image
image

### CODE REVIEW

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @zenseii, I left several comments here. Would you mind to take a look at them?

**Review by ihhub — APPROVED** [review]

**Comment by Branikolog** [comment]
Hi, @zenseii !
It would be nice, if we still support original buttons from the russian version of the game.
image

**Comment by Branikolog** [comment]
Hi, @zenseii !
Everything is ok now! Thanks for quick fix!

**Comment by Districh-ru** [comment]
Hi, @zenseii!
Yes, everything is OK! 👍 

<details>

изображение

</details>

**Comment by Districh-ru** [comment]
@zenseii,
I don't know if there are other translated Heroes 2 resources besides Russian or Polish. If there are, maybe we should support its buttons.
And should we replace button text for English language when using English resources?
So maybe do this check:
```cpp
    bool isSameCurrentLanguageAndResourceFile()
    {
        return fheroes2::getCurrentLanguage() == fheroes2::getResourceLanguage();
    }

```
What do you think? :)

**Review by Branikolog — APPROVED** [review]

File: src/fheroes2/agg/agg_image.cpp [hunk 1]
```diff
@@ -98,6 +108,12 @@ namespace
     }
 #endif
 
+    bool isPolishOrRussianLanguageAndResources()
+    {
+        return ( fheroes2::getCurrentLanguage() == fheroes2::SupportedLanguage::Polish && fheroes2::getResourceLanguage() == fheroes2::SupportedLanguage::Polish )
+               || ( fheroes2::getCurrentLanguage() == fheroes2::SupportedLanguage::Russian && fheroes2::getResourceLanguage() == fheroes2::SupportedLanguage::Russian );
+    }
+
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:116]: We should name this function more generically like `useOriginalResources`. We should also cache function's return values as they aren't inlined:
```cpp
bool useOriginalResources()
{
    const fheroes2::SupportedLanguage currentLanguage = fheroes2::getCurrentLanguage();
    const fheroes2::SupportedLanguage resourceLanguage = fheroes2::getResourceLanguage();

    return ( currentLanguage == fheroes2::SupportedLanguage::Polish && resourceLanguage == fheroes2::SupportedLanguage::Polish )
           || ( currentLanguage == fheroes2::SupportedLanguage::Russian && resourceLanguage == fheroes2::SupportedLanguage::Russian );
}
```

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @zenseii , I left one more comment. Can you please take a look?

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@zenseii , thank a lot for making this improvement!

**Pull request merged**